Updating guide for Creating a Log Integration (#20770)
DataDog/documentation: content/en/developers/integrations/log_integration.md

5. Click **Create**.

You can add processors within your pipelines to restructure your data and generate attributes.

**Requirements:**

- Use the [date remapper][4] to define the official timestamp for logs.
- Use a status remapper to remap the `status` of a log or a [category processor][19] for status mapped to a range (Like HTTP status codes)
- Use the attribute [remapper][5] to remap attribute keys to standard [Datadog attributes][6]. For example, an attribute key that contains the client IP must be remapped to `network.client.ip` so Datadog can display Technology Partner logs in out-of-the-box dashboards. Remove original attributes when remapping by using `preserveSource:false` to avoid duplicates.
- Use the [service remapper][7] to remap the `service` attribute or set it to the same value as the `source` attribute.
- Use the [grok processor][8] to extract values in the logs for better searching and analytics. To keep optimal performances, Grok parser must be specific (Avoid wildcard matches).
- Use the [message remapper][9] to define the official message of the log and make certain attributes searchable by full text.

For a list of all log processors, see [Processors][10].

> **Tip:** This [free course][20] provides a great overview on how to write processors and leverage standard attributes. 

Reviewer: ```suggestion
**Tip**: Take the free course [Going Deeper with Logs Processing][20] for an overview on writing processors and leveraging standard attributes. 
```

**Note:** You do not need to create facets for standard attributes. Facets for standard attributes will be added automatically by Datadog when the log pipeline is published. See the [default standard attribute list][6] for the standard Datadog attributes under their specific groups.

To easily navigate the facet list, facets are grouped together. For fields specific to the integration logs, create a single group with the same name as the `source` tag. 

1. In the log panel, click the Cog icon next to the attribute that you want in the new group.
2. Select **Edit facet/measure for @attribute**. If there isn't a facet for the attribute yet, select **Create facet/measure for @attribute**.
3. Click **Advanced options**.
4. In the **Group** field, enter the name and a description of the new group, and select **New group**.
5. Click **Update**.

**Requirements:**

> - Use standard attributes as much as possible. Otherwise, use a specific namespace (such as the integration name) to avoid conflict as facets are shared across Datadog.

Reviewer: ```suggestion
- Use standard attributes as much as possible. Otherwise, use a specific namespace (such as the integration name) to avoid conflict, because facets are shared across Datadog.
```

> - A facet has a source, it can be `log` for attributes or `tag` for tags.

Reviewer: ```suggestion
- A facet has a source. It can be `log` for attributes or `tag` for tags.
```

> - A facet has a type (String, Boolean, Double or Integer) which matches the type of the attribute. If the type of the value of the attribute does not match the one of the facet, the attribute will not be indexed with the facet.

Reviewer: ```suggestion
- A facet has a type (String, Boolean, Double or Integer) which matches the type of the attribute. If the type of the value of the attribute does not match the one of the facet, the attribute is not indexed with the facet.
```

> - Double and Integer facets can have a unit. It is composed of a family (time, bytes, …) and of a name (millisecond, gibibyte, …).

Reviewer: ```suggestion
- Double and Integer facets can have a unit. Units are composed of a family (such as time or bytes) and of a name (such as millisecond or gibibyte).
```

- A facet is stored in groups and has a description.

> - If you remapped an attribute and kept both, define a facet on a single one.

Reviewer: ```suggestion
- If you remap an attribute and keep both, define a facet on a single one.
```

## Review and deploy the integration

Datadog reviews the log integration based on the guidelines and requirements documented on this page and provides feedback to the Technology Partner. In turn, the Technology Partner reviews and makes changes accordingly.

> To start a review process, contact us through the [Datadog Partner Portal][] with your finalized pipeline, processors, and facets definition. 

Reviewer: ```suggestion
To start a review process, contact Datadog through the [Datadog Partner Portal][21] with your finalized pipeline, processors, and facets definition. 
```
